src/client/Web3SwapClient.ts

```diff
@@ -51,6 +51,23 @@ export class Web3SwapClient {
         }
     }
 
+    public async isValidOrder(dbOrder: DbOrder) {
+        let isValid = false;
+        const contract = this.contracts[dbOrder.chainId];
+        if (!contract) {
+            return Promise.resolve(isValid);
+        }
+        try {
+            isValid = await contract.check(
```

Comment:
`check` returns `[BigNumber, string[]]`. So you need to check if BigNumber is 0. Or use the `checkResultToErrors` from @airswap/utils.

The way I implemented it in marketplace:

https://github.com/airswap/airswap-marketplace/blob/8804e85c45aff27ed0a7dbf9d7b968531eda3d97/src/redux/stores/orders/ordersApi.ts#L103

So in your code I would do it like this:

```
const [count, checkErrors] = await Swap.getContract(provider.getSigner(), chainId).check(senderWallet, order);
const errors = (count && checkErrors) ? checkResultToErrors(checkErrors, count) as SwapError[] : [];

return !!errors.length
```

Reply:
Fixed thank you :D